lib/hydra/derivatives/processors/document.rb

```diff
@@ -2,27 +2,41 @@ module Hydra::Derivatives::Processors
   class Document < Processor
     include ShellBasedProcessor
 
-    def self.encode(path, _options, output_file)
-      format = File.extname(output_file).sub('.', '')
-      outdir = File.dirname(output_file)
+    def self.encode(path, format, outdir)
       execute "#{Hydra::Derivatives.libreoffice_path} --invisible --headless --convert-to #{format} --outdir #{outdir} #{path}"
     end
 
-    def encode_file(file_suffix, options = {})
-      new_output = ''
-      if file_suffix == 'jpg'
-        temp_file = File.join(Hydra::Derivatives.temp_file_base, [directives.fetch(:label).to_s, 'pdf'].join('.'))
-        new_output = File.join(Hydra::Derivatives.temp_file_base, [File.basename(temp_file).sub(File.extname(temp_file), ''), file_suffix].join('.'))
-        self.class.encode(source_path, options, temp_file)
-        self.class.encode(temp_file, options, output_file(file_suffix))
-        File.unlink(temp_file)
-      else
-        self.class.encode(source_path, options, output_file(file_suffix))
-        new_output = File.join(Hydra::Derivatives.temp_file_base, [directives.fetch(:label).to_s, file_suffix].join('.'))
-      end
-      out_file = File.open(new_output, "rb")
-      output_file_service.call(out_file, directives)
-      File.unlink(out_file)
+    # Converts the document to the format specified in the directives hash.
+    # TODO: file_suffix and options are passed from ShellBasedProcessor.process but are not needed.
+    #       A refactor could simplify this.
+    def encode_file(_file_suffix, _options = {})
+      convert_to_format
+      File.unlink(converted_file)
```

Comment:
Should we pop an `ensure` here to make sure the deletion happens, even if `convert_to_format` bombs out?